meteor/server/api/peripheralDevice.ts

```diff
@@ -609,6 +610,95 @@ PickerPOST.route('/devices/:deviceId/uploadCredentials', (params, req: IncomingM
 	res.end(content)
 })
 
+PickerGET.route('/devices/:deviceId/oauthResponse', (params, req: IncomingMessage, res: ServerResponse) => {
+	res.setHeader('Content-Type', 'text/plain')
+
+	let content = ''
+	try {
+		const deviceId: PeripheralDeviceId = protectString(decodeURIComponent(params.deviceId))
+		check(deviceId, String)
+
+		if (!deviceId) throw new Meteor.Error(400, `parameter deviceId is missing`)
+
+		const peripheralDevice = PeripheralDevices.findOne(deviceId)
+		if (!peripheralDevice) throw new Meteor.Error(404, `Peripheral device "${deviceId}" not found`)
+
+		if (!peripheralDevice.studioId)
+			throw new Meteor.Error(400, `Peripheral device "${deviceId}" is not attached to a studio`)
+
+		const studio = Studios.findOne(peripheralDevice.studioId)
+		if (!studio) throw new Meteor.Error(404, `Studio "${studio}" not found`)
+
+		const url = new URL(req.url || '', 'http://localhost')
+
+		let accessToken = url.searchParams.get('code') || undefined
+		const scopes = url.searchParams.get('scope') || undefined
+
+		check(accessToken, String)
+		check(scopes, String)
+
+		accessToken = (accessToken + '').trim()
+		if (accessToken && accessToken.length > 5) {
+			PeripheralDeviceAPI.executeFunction(deviceId, 'receiveAuthToken', accessToken)
```

Comment:
Is it intentional that we do the 302 redirect even if this executeFunction fails? A comment to state this and why would be good